addons/sale_loyalty/tests/test_program_with_code_operations.py

```diff
@@ -411,3 +431,143 @@ def test_edit_and_reapply_promotion_program(self):
         self._apply_promo_code(order, 'test')
         # But the above line should not add any reward
         self.assertEqual(len(order.order_line), 2, "You should get a discount line") # product + discount
+
+    def test_reapply_multiple_global_rewards_when_new_discount_greater(self):
+        """ Test applying the maximum reward discount from multiple rewards when the applied
+            coupon discount is lower.
+
+        1. Create a two loyalty program of type `coupons`.
+        2. Add multiple rewards to the second program.
+        3. Generate a coupon for each program.
+        2. Create a sale order and add a product to it.
+        3. Apply the Coupon to the sale order.
+        4. Try to apply the second coupon with multiple rewards.
+        5. Reward with best discount will be shown.
+        """
+        self.code_promotion_program_with_discount.rule_ids.unlink()
+        coupon_1 = self._generate_coupons(self.code_promotion_program_with_discount)
+        coupon_2 = self._generate_coupons(self.discount_with_multi_rewards)
+
+        order = self.empty_order
+        self.assertEqual(order.amount_total, 0.0)
+        order.write({'order_line': [
+            Command.create({
+                'product_id': self.product_A.id,
+                'name': '1 Product A',
+                'product_uom': self.uom_unit.id,
+                'product_uom_qty': 1.0,
+            }),
+            Command.create({
+                'product_id': self.product_B.id,
+                'name': '1 Product B',
+                'product_uom': self.uom_unit.id,
+                'product_uom_qty': 1.0,
+            }),
+        ]})
+
+        # The order line should be created with the correct price unit
+        self.assertEqual(len(order.order_line.ids), 2)
+        self.assertEqual(order.order_line[0].price_unit, 100.0)
+        self.assertEqual(order.order_line[1].price_unit, 5.0)
+        expected_total = order.amount_total * 0.80
+
+        # Apply the first coupon
+        self._apply_promo_code(order, coupon_1.code)
+        self.assertEqual(len(order.order_line.ids), 3)
+        self.assertEqual(order.order_line[2].price_unit, -10.5, "Order total should reflect the 10% discount")
+
+        # Apply the second coupon with multiple rewards
+        self._apply_promo_code(order, coupon_2.code)
+        self.assertEqual(len(order.order_line.ids), 3)
+        self.assertEqual(order.order_line[2].price_unit, -21.0, msg="Order total should reflect the 20% discount")
+        self.assertAlmostEqual(order.amount_total, expected_total, msg="Order total should reflect the 20% discount")
+
+    def test_choose_best_discount_when_reapply_multiple_global_rewards__higher(self):
```

Comment:
```suggestion
    def test_choose_best_discount_when_reapply_multiple_global_rewards_higher(self):
```
Or, to show the difference with the previous one:
```suggestion
    def test_reapply_multiple_higher_global_rewards_lets_choose_best(self):
```